### PR TITLE
fix issue CAT-1061

### PR DESCRIPTION
**Pull request by qianlvable**
fix by when insert operator into number， divide the number into two new number token

### CODE REVIEW

**Comment by obusher** [comment]
Please rebase onto `Catrobat:develop` and [rewrite](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase-i) your commit-message.

**Comment by avihric** [comment]
@qianlvable Hey, hope you did not lose your motivation. Sorry for the delay, but we had really big problems to stabilze our  master branch. Could you rebase and rewrite your commit message like @obusher said.

Thanks in advance.

**Comment by qianlvable** [comment]
Hey,Ajdin!
Really sorry for not being active for a long time,too many thing to do at
school.I will fix this tomorrow.Thanks for your patient.
Regards
Nick

2015-03-30 20:36 GMT+08:00 Ajdin Vihric <notifications@github.com>:

**Comment by qianlvable** [comment]
Hi,@avihric ,@obusher. Sorry again for delay so long.When I try to rebase use follow command "git checkout -b fix-cat1061 master" and "git rebase develop" and then "git merge fix-cat1061". And my git history shows below:

Notice that it seems like I mix it with another commit CAT826(another issue that I work on).How could I fix this?

**Comment by avihric** [comment]
You forgot to pull the latest changes from origin develop.
So I would do the following:
* pull the latest changes from upstream develop
* switch to your master branch and rebase to upstream develop
* rewrite your commit message to: fix issue CAT-1061

**Comment by qianlvable** [comment]
Hi,@avihric
I have tried to pull and rebase,this is now my git history look:
test
It seems like "fix issue 826" this commit become the head of develop branch. How could I commit the "fix-1061" only?
I try to use git checkout the"fix issue 1061"commit,but the git shows me this message "You are in 'detached HEAD' state. You can look around, make experimen changes and commit them, and you can discard any commits you make in state without impacting any branches by performing another checkout."

**Comment by obusher** [comment]
I assume you are on branch `develop`:
* `git log` and copy your commit hash of `fix issue CAT-1061`
* `git checkout master`
* `git reset --hard upstream/develop`
* `git cherry-pick <commit hash of fix issue CAT-1061>`
You should see now with `git log` that you successfully picked the commit.

Rewrite your commit message with:
`git commit --amend`

`git push -f` your changes. :octopus:

**Comment by qianlvable** [comment]
@obusher You are so awesome!Work like magic,and I think I really need to put more effort to study git : ) Thx

**Comment by qianlvable** [comment]
@obusher  After following your instruction,"fix-cat1061"commit become in the master,it did not appear in develop branch.
test

So I switch to "develop" and use command " git rebase master",now my git history look like this:

It looks more and more confused : ( 
Thanks for you patient

**Comment by obusher** [comment]
Your local `develop` should be mirror of `Catrobat:develop`.
You should not work on this branch, only fetch and/or pull updates from `Catrobat:develop`.
There was no need to rebase your local `develop` onto `master` :P
Stay on your `master` for this issue (PR)! 

Please reformat your code (Ctrl + Alt + L) but only those lines which you are contributing.
Amend your changes to the previous commit and push them. :fallen_leaf:

**Comment by qianlvable** [comment]
Work like magic,@obusher  you are awesome! thx

**Comment by obusher** [comment]
Your welcome! :)
Please do not forget to add your changes prior commiting.  (`git add` < your files >) :taxi:

**Comment by qianlvable** [comment]
Hi,@obusher and @avihric,this build failed seems not related the code that I change, is that right?(I have run test successfully before I rebasing the code : ( 
I try to test on my computer,when click the edittext,this error shows up "  java.lang.NullPointerException
            at org.catrobat.catroid.ui.fragment.FormulaEditorFragment.onPrepareOptionsMenu(FormulaEditorFragment.java:345)"
Do I need to look into it?

File: catroid/src/org/catrobat/catroid/formulaeditor/InternFormula.java [hunk 1]
```diff
@@ -736,6 +736,17 @@ private CursorTokenPropertiesAfterModification replaceCursorPositionInternTokenB
 
 		Log.i("info", "replaceCursorPositionInternTokenByTokenList:enter");
 
+		if (cursorPositionInternToken.isNumber() && internTokensToReplaceWith.size() == 1
+				&& internTokensToReplaceWith.get(0).isOperator()){
+
+			int externNumberOffset = externInternRepresentationMapping.getExternTokenStartOffset(externCursorPosition,
+					cursorPositionInternTokenIndex);
+			List<InternToken> replaceList = InternFormulaUtils.insertOperatorToNumberToken(cursorPositionInternToken,externNumberOffset,internTokensToReplaceWith.get(0));
```
Comment (obusher) [inline catroid/src/org/catrobat/catroid/formulaeditor/InternFormula.java]: Your code is still not formated. No whitespace after parameter-delimiter `, ` .

**Comment by obusher** [comment]
Try to reproduce this `NullPointerExeption` on the current Catrobat:develop branch.

**Comment by qianlvable** [comment]
I reformat my code.And I checkout to the "upstream/develop"  branch and try to run same error occur : (

**Comment by obusher** [comment]
Please create a issue on our [Jirra server](https://jira.catrob.at/).

**Comment by obusher** [comment]
Please [squash](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase-i/) your commits into a single one.

**Pull request merged**